apiserver/facades/client/client/status.go

```diff
@@ -574,12 +590,18 @@ func fetchAllApplicationsAndUnits(
 			continue
 		}
 		if err != nil {
-			return nil, nil, nil, err
+			return allApplicationsUnitsBindingsCharmStatus{}, err
 		}
 		latestCharms[baseURL] = ch
 	}
 
-	return appMap, unitMap, latestCharms, nil
+	var allAppsUnitsChamrBindings allApplicationsUnitsBindingsCharmStatus
+	allAppsUnitsChamrBindings.applications = appMap
+	allAppsUnitsChamrBindings.units = unitMap
+	allAppsUnitsChamrBindings.latestCharms = latestCharms
+	allAppsUnitsChamrBindings.endpointBindings = allBindingsByApp
+
+	return allAppsUnitsChamrBindings, nil
```

Comment:
typically we construct this inline:

return applicationStatusInfo {
applications: appMap,
units: unitMap,
latestCharms: ;atestCharms,
endpointBindings: allBindingsByApp,
}, nil